wagtail_streamfield_migration_toolkit/autodetect/streamchangedetector.py

```diff
@@ -99,31 +100,46 @@ def find_renamed_or_removed_defs(
             old_child_def = old_child_defs[old_child_name]
             is_child_mapped = False
 
-            # For now assume that same name means same block (i.e. the old block wasn't deleted
-            # and a new block with the same name added)
+            # Find out if the block maps to one of the new only children. If it maps, that
+            # would mean that the block has been renamed
+
+            comparer = block_def_comparer_registry.get_block_def_comparer_for_instance(
+                old_child_def
+            )
             if old_child_name in new_child_names:
-                new_child_def = new_child_defs[old_child_name]
-                # recursion call
-                self.find_renamed_or_removed_defs(
-                    old_child_def,
-                    new_child_def,
-                    old_child_path,
+                new_child_name = old_child_name
+                new_child_def = new_child_defs[new_child_name]
+                new_child_path = parent_path + path_suffix + new_child_name
```

Comment:
Perhaps this is just semantics, but `separator` is more accurate and in line with string processing conventions than `suffix`.